third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -890,10 +938,26 @@ impl ContainerRef {
                         };
                         ValueImpl::ContainerRef(r)
                     },
-                    _ => ValueImpl::IndexedRef(IndexedRef {
+
+                    ValueImpl::U8(_)
+                    | ValueImpl::U16(_)
+                    | ValueImpl::U32(_)
+                    | ValueImpl::U64(_)
+                    | ValueImpl::U128(_)
+                    | ValueImpl::U256(_)
+                    | ValueImpl::Bool(_)
+                    | ValueImpl::Address(_)
+                    | ValueImpl::DelayedFieldID { .. } => ValueImpl::IndexedRef(IndexedRef {
                         idx,
                         container_ref: self.copy_value(),
                     }),
+
+                    ValueImpl::ContainerRef(_) | ValueImpl::Invalid | ValueImpl::IndexedRef(_) => {
```

Comment:
Doesn't this change behavior? (although not sure how it would have worked in the past). Do we need replay (or it would be invariant violation somewhere in the caller and unchanged) or a test?

Reply:
It's an invariant violation so if things are implemented correctly it should have been reachable in the past.

Reply:
We cannot have references stored in containers, and there was a TODO for that above, so these cases should not have been reachable (cc @runtian-zhou to confirm). I will start a replay once all comments are addressed.